compiler/rustc_errors/src/diagnostic.rs

```diff
@@ -108,6 +108,15 @@ impl IntoDiagnosticArg for char {
     }
 }
 
+impl<T: IntoDiagnosticArg> IntoDiagnosticArg for Option<T> {
```

Comment:
What is this used for? I'm not sure if we want to have this as a general implementation, since the meaning of `Option<T>` in many different diagnostics is very different.

Reply:
Oh, it's that `Option<i32>`? I think you should make a newtype wrapper around `Option<i32>` and implement `IntoDiagnosticArg` in the crate instead of in `rustc_errors`.